cpp/include/cudf/table/experimental/row_operators.cuh

```diff
@@ -114,6 +114,161 @@ using lhs_iterator = strong_index_iterator<lhs_index_type>;
+   * @return Indicates the relationship between the elements in the `lhs` and `rhs` columns, along
+   * with the depth at which a null value was encountered.
+   */
+  template <typename Element, CUDF_ENABLE_IF(cudf::is_relationally_comparable<Element, Element>())>
+  __device__ cuda::std::pair<weak_ordering, int> operator()(
+    size_type const lhs_element_index, size_type const rhs_element_index) const noexcept
+  {
+    if (_check_nulls) {
+      bool const lhs_is_null{_lhs.is_null(lhs_element_index)};
+      bool const rhs_is_null{_rhs.is_null(rhs_element_index)};
+
+      if (lhs_is_null or rhs_is_null) {  // at least one is null
+        return cuda::std::pair(null_compare(lhs_is_null, rhs_is_null, _null_precedence), _depth);
+      }
+    }
+
+    return cuda::std::pair(relational_comparator(_lhs.element<Element>(lhs_element_index),
+                                                 _rhs.element<Element>(rhs_element_index)),
+                           std::numeric_limits<int>::max());
+  }
+
+  template <typename Element,
+            CUDF_ENABLE_IF(not cudf::is_relationally_comparable<Element, Element>() and
+                           not std::is_same_v<Element, cudf::struct_view>)>
+  __device__ cuda::std::pair<weak_ordering, int> operator()(size_type const,
+                                                            size_type const) const noexcept
+  {
+    CUDF_UNREACHABLE("Attempted to compare elements of uncomparable types.");
+  }
+
+  template <typename Element, CUDF_ENABLE_IF(std::is_same_v<Element, cudf::struct_view>)>
+  __device__ cuda::std::pair<weak_ordering, int> operator()(
+    size_type const lhs_element_index, size_type const rhs_element_index) const noexcept
+  {
+    column_device_view lcol = _lhs;
+    column_device_view rcol = _rhs;
+    int depth               = _depth;
+    while (lcol.type().id() == type_id::STRUCT) {
+      bool const lhs_is_null{lcol.is_null(lhs_element_index)};
+      bool const rhs_is_null{rcol.is_null(rhs_element_index)};
+
+      if (lhs_is_null or rhs_is_null) {  // at least one is null
+        weak_ordering state = null_compare(lhs_is_null, rhs_is_null, _null_precedence);
+        return cuda::std::pair(state, depth);
+      }
+
+      if (lcol.num_child_columns() == 0) {
+        return cuda::std::pair(weak_ordering::EQUIVALENT, depth);
+      }
+
+      // Non-empty structs have been modified to only have 1 child when using this.
+      lcol = detail::structs_column_device_view(lcol).get_sliced_child(0);
+      rcol = detail::structs_column_device_view(rcol).get_sliced_child(0);
+      ++depth;
+    }
+
+    auto const comparator = element_comparator{_check_nulls, lcol, rcol, _null_precedence, depth};
+    return cudf::type_dispatcher<dispatch_void_if_nested>(
+      lcol.type(), comparator, lhs_element_index, rhs_element_index);
+  }
+
+  template <typename Element, std::enable_if_t<not std::is_floating_point_v<Element>>* = nullptr>
```

Comment:
We should use `CUDF_ENABLE_IF` consistently.

The same for other use cases in this PR.